### DISCUSSION TITLE
.cfg setup

### DISCUSSION THREAD

**Discussion by gonPLD** [NAnand-TUD/parablade#16] [Q&A]
Hi! 

I am currently performing a 3D turbine blade design and I got the .cfg from the parablade examples to not start from scratch. I have several soubts though about certain parameters and nomenclature:

- Do the variables have units? Angles are clearly in degrees, but the rest of them I do not know if there are non-dimensional parameters or they are in mm, m, in, etc (for instance, z_shroud, dist_1...)

- The parameters are expressed with three values. Do theuy correspond to different span locations or how are they used if not?

- Finally, I would like to know what are dist_1, dist_2, ... , dist_in, dist_out and how you define stagger angle in your program, is it the angle between the direction of rotation and LE-TE line?

Thanks a lot in advance!
Gonzalo

**Comment by NAnand-TUD** [comment]
Hi Gonzalo,

Good to hear that you are using ParaBlade is it of some use to you.

To start with a 3D Geom the best is to start with a 3D example in the test cases and adapt it to your case. Please note that there is a PlotBlade.py executable using which you can have interactive plots and see live changes in the parameters. This way it will be easier to understand what is happening with each parameter. 

Yes, for 3D more variables you have will define more control points for control in the 3rd direction.

Regarding dist, please refer to our paper in CAD-Journal. I am sure it will help you understand well what these parameters mean. Link: https://www.semanticscholar.org/paper/A-Unified-Geometry-Parametrization-Method-for-Agromayor-Anand/1554b56266dcfb1ee6dbabb095ade44f8c4376bd

Please do not hesitate to put your questions if you have more doubts.

Thanks,
Nitish
